### PR TITLE
[CALCITE-4783] RelFieldTrimmer incorrectly drops filter condition

### PR DESCRIPTION
**Pull request by shyzeshzye**
fix RelFieldTrimmer incorrectly drops calc condition when there is no field used by the upper 
https://issues.apache.org/jira/browse/CALCITE-4783

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql2rel/RelFieldTrimmer.java [hunk 1]
```diff
@@ -371,6 +371,15 @@ public TrimResult trimFields(
     final List<RexNode> projs = Util.transform(rexProgram.getProjectList(),
         rexProgram::expandLocalRef);
 
+    RexNode conditionExpr = null;
+    if (rexProgram.getCondition() != null) {
+      final List<RexNode> filter = Util.transform(
+          ImmutableList.of(
+              rexProgram.getCondition()), rexProgram::expandLocalRef);
+      assert filter.size() == 1;
```
Comment (xy2953396112) [inline core/src/main/java/org/apache/calcite/sql2rel/RelFieldTrimmer.java:379]: Why `filter.size() == 1`?

**Comment by xy2953396112** [comment]
Please not use `fix` for PR title.

File: core/src/main/java/org/apache/calcite/sql2rel/RelFieldTrimmer.java (hunk 1 above)
Comment (shyzeshzye) [reply core/src/main/java/org/apache/calcite/sql2rel/RelFieldTrimmer.java:379]: RexProgram has only one condition, so  it means rexProgram.getCondition() != null.

**Review by chunweilei — APPROVED** [review]
LGTM

**Pull request merged**